File: src/main/java/com/koddy/server/file/domain/model/DefaultBucketFileNameGenerator.java
package com.koddy.server.file.domain.model;

import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class DefaultBucketFileNameGenerator implements BucketFileNameGenerator {
    @Override
    public String get(final String fileName) {
        final FileExtension extension = FileExtension.getExtensionViaFimeName(fileName);
        return UUID.randomUUID() + extension.getValue();
    }
}

File: src/main/kotlin/com/koddy/server/file/domain/model/DefaultBucketFileNameGenerator.kt
package com.koddy.server.file.domain.model

import org.springframework.stereotype.Component
import java.util.UUID

@Component
class DefaultBucketFileNameGenerator : BucketFileNameGenerator {
    override fun get(fileName: String): String = UUID.randomUUID().toString() + FileExtension.from(fileName).value
}
